[CI] RestCancellableNodeClientTests testConcurrentExecuteAndClose failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #9081 / almalinux-8-aarch64_platform-support-arm](https://gradle-enterprise.elastic.co/s/4nvyqgctajmjy)
- [elasticsearch-periodic-platform-support #9059 / rhel-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/3dgnhllgzop5q)

**Reproduction Line:**
```
./gradlew ':server:test' --tests "org.elasticsearch.rest.action.RestCancellableNodeClientTests.testConcurrentExecuteAndClose" -Dtests.seed=5D8A75F24A010EF6 -Dtests.locale=da -Dtests.timezone=Etc/GMT+0 -Druntime.java=22
```

**Applicable branches:**
7.17

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.rest.action.RestCancellableNodeClientTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testConcurrentExecuteAndClose),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected:<25> but was:<26>
```

**Issue Reasons:**
- [7.17] 2 failures in test testConcurrentExecuteAndClose (1.5% fail rate in 136 executions)
- [7.17] 2 failures in pipeline elasticsearch-periodic-platform-support (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -5,6 +5,9 @@ tests:
 - class: org.elasticsearch.packaging.test.CertGenCliTests
   method: test40RunWithCert
   issue: https://github.com/elastic/elasticsearch/issues/109831
+- class: org.elasticsearch.rest.action.RestCancellableNodeClientTests
+  method: testConcurrentExecuteAndClose
+  issue: https://github.com/elastic/elasticsearch/issues/130372
 
 # Examples:
 #

